minimark/src/rmrk/types.ts

```diff
@@ -11,20 +11,70 @@ export enum Interaction {
 }
 
 export type JustInteraction = Exclude<Interaction, Interaction.MINT | Interaction.MINTNFT | Interaction.UNLIST>
+export enum InteractionV2 {
+  ACCEPT = 'ACCEPT',
+  BASE = 'BASE',
+  EQUIP = 'EQUIP',
+  EQUIPPABLE = 'EQUIPPABLE',
+  LOCK = 'LOCK',
+  RESADD = 'RESADD',
+  SETPROPERTY = 'SETPROPERTY',
+  SETPRIORITY = 'SETPRIORITY',
+  THEMEADD = 'THEMEADD',
+}
 
 export type OnlyMintInteraction = Interaction.MINT | Interaction.MINTNFT
 
+// type JSONFiled = Record<string, string> | undefined
+
 export type UnwrappedRemark<T> = {
   interaction: Interaction
   value: T
   version: string
 }
 
+export type UnwrappedRemark2<T> = {
+  interaction: InteractionV2
+  value: T
+  version: string
+}
+
 export type InteractionValue = {
   id: string
   value?: string
 }
 
+export type Base = {
+  value: any
+}
+
+export type Equip = Omit<InteractionValue, 'value'> & {
+  baseslot: string
+}
+
+export type Equippable = Required<InteractionValue> & {
+  slot: string
+}
+
+export type Lock = Omit<InteractionValue, 'value'>
+
+export type Resadd = Required<InteractionValue> & {
+  value: Record<string, string>
+  replace: string
+}
+
+export type Setpriority = Required<InteractionValue>
+
+export type Setproperty = Required<InteractionValue> & {
+  name: string
+}
+
+export type Themeadd = Required<InteractionValue> & {
```

Comment:
```suggestion
export type AddTheme = Required<InteractionValue> & {
```

Reply:
This is what the standard is named, in order to consistency, I would revert that to `ThemeAdd`[1]

https://github.com/rmrk-team/rmrk-spec/blob/master/standards/rmrk2.0.0/interactions/themeadd.md